### PR TITLE
Refactor cover art retrieval

### LINKED ISSUE #128: Album art is not displayed
There are failing requests to last.fm for getting the album art. The returned message is always like:
```
    jsonp144560929891147({"error":6,"message":"Album not found","links":[]})
```
This seems to be the case because the request is missing the required artist name:
```
    http://ws.audioscrobbler.com/2.0/?artist=&album=The+Better+Life&method=album.getInfo&api_key=...&callback=jsonp144560929891147&format=json
```
Probably this was caused by a last.fm API change of [album.getInfo](http://www.last.fm/api/show/album.getInfo).

### PR DESCRIPTION
**Pull request by jcass77**
I've refactored the Javascript routines for retrieving album covers:

- Optimised to pass track URI's instead of full album models.
- Now uses the new ``mopidy.library.getImages`` convention before falling back to the deprecated ``album.images`` approach.
- Now also falls back to using track artist when doing lookups on last.fm if album artist is not available (fixes #128).

Just for the avoidance of doubt, the order of precedence is: ``mopidy.library.getImages`` -> ``album.images`` -> last.fm

### CODE REVIEW

**Comment by jcass77** [comment]
I think I'm satisfied to merge this if there are no further comments or inputs received in the next 24 hours?

**Comment by kingosticks** [comment]
Yeh I will give this a test later today.

File: mopidy_musicbox_webclient/static/js/functionsvars.js [hunk 1]
```diff
@@ -343,12 +343,9 @@ function resultsToTables(results, target, uri) {
                     }
                     newalbum = [];
                     newtlids = [];
-                    if (results[i].album) {
-                        coversList.push([results[i].album, i]);
-                    }
                 } //newalbum length
                 if (results[i].album) {
-                    coversList.push([results[i].album, i]);
+                    coversList.push([results[i].uri, i]);
```
Comment (kingosticks) [inline mopidy_musicbox_webclient/static/js/functionsvars.js]: Did you mean to move this outside the `if`? In the case of there being only one track in the album, the cover art is not displayed. Or is it requested anyway to be used in another view? I admit I'm not 100% familiar with this code.

Comment (kingosticks) [inline mopidy_musicbox_webclient/static/js/functionsvars.js]: No, it appears it's not used anywhere else. All other places where we display cover art we seem to re-request the image anyway.

File: mopidy_musicbox_webclient/static/js/functionsvars.js [hunk 2]
```diff
@@ -356,7 +353,7 @@ function resultsToTables(results, target, uri) {
     tableid = "#" + tableid;
     $(target).html(html);
     $(target).attr('data', uri);
-    //retreive albumcovers
+    //retrieve albumcovers
     for (i = 0; i < coversList.length; i++) {
         getCover(coversList[i][0], target + '-cover-' + coversList[i][1], 'small');
```
Comment (kingosticks) [inline mopidy_musicbox_webclient/static/js/functionsvars.js]: Out of scope for now maybe, but do you have plans to group these into a single `getImages` request at some point?

File: mopidy_musicbox_webclient/static/js/functionsvars.js (hunk 1 above)
Comment (jcass77) [inline mopidy_musicbox_webclient/static/js/functionsvars.js]: I couldn't see why it was necessary to have the same statement inside the ``if``, and then again immediately outside it. Did I miss something there?

File: mopidy_musicbox_webclient/static/js/functionsvars.js (hunk 2 above)
Comment (jcass77) [inline mopidy_musicbox_webclient/static/js/functionsvars.js]: The whole ``resultsToTables`` function is a little hard to follow and probably due for some refactoring (I'm not sure where to start though).

Until I have a better understanding of what it does and how it works: I'm not sure that a ``getImages`` will be useful anywhere else. But I did notice that there are a lot of calls go ``getCover`` for just about every navigation action in the client, so doing it once and caching the results for retrieval later on would most likely make more sense.

Comment (kingosticks) [inline mopidy_musicbox_webclient/static/js/functionsvars.js]: Just to be clear, I meant the optimisation of calling mopidy.getImages just once, but with all the covers to fetch. Currently we do individual calls for each cover.

File: mopidy_musicbox_webclient/static/js/functionsvars.js (hunk 1 above)
Comment (kingosticks) [inline mopidy_musicbox_webclient/static/js/functionsvars.js]: Oh, I didn't realise it was previously in *both* places. Surely once, inside the if statement should be sufficient. I agree it's a beast of a function and I've always steered clear but looking last night it's not too bad. It's just trying to split the results up by album so it can insert an album row (but only in the case where there are multiple consecutive  tracks for the album).

File: mopidy_musicbox_webclient/static/js/functionsvars.js (hunk 2 above)
Comment (jcass77) [inline mopidy_musicbox_webclient/static/js/functionsvars.js]: Oh now I get it. Event so, at the moment only the call from ``resultsToTables`` will have more than one URI available for calling `` mopidy.getImages`` on, so we can probably leave that for another PR as you suggested.

I'm still a little confused as to why the images are being retrieved here to start with, as some of the tables being rendered in the client don't even have the ability to display images as far as I can tell.

Comment (kingosticks) [inline mopidy_musicbox_webclient/static/js/functionsvars.js]: I didn't notice that, which ones don't display images?

**Comment by jcass77** [comment]
Anything else that needs to happen here before merging?

File: mopidy_musicbox_webclient/static/js/functionsvars.js (hunk 1 above)
Comment (kingosticks) [inline mopidy_musicbox_webclient/static/js/functionsvars.js]: I'd suggest moving this inside the if

File: mopidy_musicbox_webclient/static/js/functionsvars.js (hunk 2 above)
Comment (jcass77) [inline mopidy_musicbox_webclient/static/js/functionsvars.js]: From what I can see, ```resultsToTables``` only shows images if an 'album header' forms part of the search results:

skitched-20160221-200155

...and I think that only happens when all of the tracks in the entire album are listed in either the search results or the play queue. Many of the extensions that I am using adds individual tracks to the queue however, so I don't often get to see album covers in the results tables.

skitched-20160221-200429

Probably nothing wrong with this, just a little surprising in terms of what I would expect personally.

File: mopidy_musicbox_webclient/static/js/functionsvars.js (hunk 1 above)
Comment (jcass77) [inline mopidy_musicbox_webclient/static/js/functionsvars.js]: ...so the idea is that we only push an album cover if more than one track is being shown for a particular album?

Done in 7d680fe252b5d4e5c28619e8ba1e67705891a0c2!

File: mopidy_musicbox_webclient/static/js/functionsvars.js (hunk 2 above)
Comment (jcass77) [inline mopidy_musicbox_webclient/static/js/functionsvars.js]: My bewilderment is probably related to https://github.com/pimusicbox/mopidy-musicbox-webclient/issues/99.

Comment (kingosticks) [inline mopidy_musicbox_webclient/static/js/functionsvars.js]: An album header should be shown when they are more than one tracks form the same album. It should not require all the tracks. I believe it has this behaviour to avoid looking stupid (and twice as long) when it is just a big list of random songs.

**Comment by kingosticks** [comment]
If you are able to just double check that moving that if statement still grabbed covers for everything that wanted one .. then LGTM.

**Comment by jcass77** [comment]
Yes I think we are good.
